Sources/Subs-Db-postgresql.php

```diff
@@ -922,4 +923,53 @@ function smf_db_escape_wildcard_string($string, $translate_human_wildcards = fal
 	return strtr($string, $replacements);
 }
 
+/**
+ * Updates data in a table, using data from other tables
+ *
+ * @param array $table Associative array with info about the table to be updated ('name' => '{db_prefix}foo', 'alias' => 'f')
+ * @param array $from Array of associative arrays with info about the other tables to get data from ('name' => '{db_prefix}bar', 'alias' => 'b', 'condition' => 'f.baz = b.qux')
+ * @param string $set A string containing the SET instructions for the update query
+ * @param string $where A string containing any WHERE conditions for the update query
+ * @param array $db_values The values to be inserted into the compiled query string
+ * @param object $connection The connection to use (if null, $db_connection is used)
+ * @return bool True if the update was successful, otherwise false
+ */
+function smf_db_update_from($table, $joined, $set, $where, $db_values, $connection = null)
+{
+	global $smcFunc, $db_connection, $db_prefix;
+
+	$connection = $connection === null ? $db_connection : $connection;
+
+	if (empty($table['name']) || empty($table['alias']) || empty($set))
+		return;
+
+	if (!empty($where))
+		$where = '(' . $where . ')';
+
+	$from = array();
+	foreach ($joined as $join)
+	{
+		if (empty($join['name']) || empty($join['alias']) || empty($join['condition']))
+			continue;
+
+		$from[] = $join['name'] . ' AS ' . $join['alias'];
+		$where = (!empty($where) ? $where .' AND ' : '') . '(' . $join['condition'] . ')';
+	}
+	if (empty($from))
+		return;
+
+	// Postgres doesn't like prefixes on the columns to be set
+	$set = preg_replace('~\b' . $table['alias'] . '\.\b~', '', $set);
```

Comment:
Table name or alias prefixes. As it says in [the Postgres documentation](https://www.postgresql.org/docs/9.1/static/sql-update.html) regarding the `column` definition for `UPDATE` syntax:

> Do not include the table's name in the specification of a target column — for example, UPDATE tab SET tab.col = 1 is invalid.